crates/oxc_traverse/src/context/scoping.rs

```diff
@@ -102,6 +102,24 @@ impl TraverseScoping {
         self.insert_scope_below(&collector.scope_ids, flags)
     }
 
+    /// Insert a scope into scope tree below a statement.
+    ///
+    /// Statement must be in provided scope.
+    /// New scope is created as child of the provided scope.
+    /// All child scopes of the statement are reassigned to be children of the new scope.
+    ///
+    /// `flags` provided are amended to inherit from parent scope's flags.
+    pub fn insert_scope_below_statement_from_scope_id(
+        &mut self,
+        stmt: &Statement,
+        scope_id: ScopeId,
+        flags: ScopeFlags,
+    ) -> ScopeId {
+        let mut collector = ChildScopeCollector::new();
+        collector.visit_statement(stmt);
+        self.insert_scope_below_from_scope_id(&collector.scope_ids, scope_id, flags)
+    }
```

Comment:
Could you please add a "shortcut" method to `TraverseCtx` too which delegates to this one? - like I did in #9960 after one of your previous PRs.